File: examples/nodeJSExamples/GDBaaS/sparql_update.js
/*
Copyright 2016 Ontotext AD

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

var https = require("https");
var querystring = require("querystring");

var str = ['PREFIX dc: <http://purl.org/dc/elements/1.1/>',
'INSERT DATA { <http://example/egbook> dc:title  "This is an example title" }'].join("\n");

var dataString = querystring.stringify({
    update: str
});

var headers = {
    "Content-Type": "application/x-www-form-urlencoded"
};

var options = {
    hostname: "rdf.s4.ontotext.com",
    path: "/<user-id>/<db>/repositories/<repository>/statements",
    method: "POST",
    auth: "<s4 api key>:<s4 key secret>",
    headers: headers
};

// Setup the request.  The options parameter is
// the object we defined above.
var req = https.request(options, function (res) {
    res.setEncoding("utf-8");

    var responseString = "";

    res.on("data", function (data) {
        responseString += data;
    });

    res.on("end", function () {
        console.log(responseString);
        console.log("\nStatus Code: ", res.statusCode);
        console.log("\nHeaders:");
        console.log(res.headers);
    });
});

req.on("error", function (e) {
    console.log(e);
});

req.end(dataString);

File: examples/PythonExamples-requests/GDBaaS/sparql_update.py
# Copyright  2013, 2014, Ontotext AD
#
# This file is free software; you can redistribute it and/or modify it under
# the terms of the GNU Lesser General Public License as published by the Free
# Software Foundation; either version 2.1 of the License, or (at your option)
# any later version.
# This library is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
# details.
# You should have received a copy of the GNU Lesser General Public License
# along with this library; if not, write to the Free Software Foundation, Inc.,
# 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA

# IMPORTANT!!! Please install the packages below, needed for this example

import requests


endpoint = ("https://rdf.s4.ontotext.com/<user-id>/<databaseName>/"
            + "repositories/<repoName>/statements")
api_key = "<your-credentials-here>"
key_secret = "<your-credentials-here>"
query = """PREFIX dc: <http://purl.org/dc/elements/1.1/>
INSERT DATA {<http://example/egbook> dc:title \"This is an example title\"}"""

headers = {
    "Content-Type": "application/x-www-form-urlencoded"
}

req = requests.post(
    endpoint, headers=headers,
    data="update="+query, auth=(api_key, key_secret))

# Response status code
print ("Request Code: {}\n".format(req.status_code))

# Response Headers
head = dict(req.headers)
print ("Headers: ")
for each in head:
    print (each.capitalize(), ": ", head[each])
